### PR TITLE
Token addition

### PR DESCRIPTION
**Pull request by BeeHiveJava**

### CODE REVIEW

File: web/src/main/java/nl/tehdreamteam/se42/web/token/TokenRegistryFactory.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package nl.tehdreamteam.se42.web.token;
+
+import nl.tehdreamteam.se42.web.token.impl.HashTokenRegistry;
+
+/**
+ * Holds a single instance of a {@link TokenRegistry}.
+ */
+public class TokenRegistryFactory {
```
Comment (TehMessiah) [inline web/src/main/java/nl/tehdreamteam/se42/web/token/TokenRegistryFactory.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AS1118)

File: web/src/main/java/nl/tehdreamteam/se42/web/soap/conversation/SoapConversationEndpoint.java [hunk 2]
```diff
@@ -14,7 +14,7 @@
  * @author Oscar de Leeuw
  */
 @WebService(targetNamespace = SoapWebServiceConstants.DEFAULT_NAMESPACE + "conversation")
-public class SoapConversationController implements ConversationController {
+public class SoapConversationEndpoint implements ConversationEndpoint {
     @Override
     public void addMessage(@WebParam(name = "token") String token, @WebParam(name = "content") String content, @WebParam(name = "conversation_id") long conversationId) {
```
Comment (ghost) [inline web/src/main/java/nl/tehdreamteam/se42/web/soap/conversation/SoapConversationEndpoint.java:19]: CRITICAL Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AS1186)

File: web/src/main/java/nl/tehdreamteam/se42/web/soap/user/SoapUserEndpoint.java [hunk 3]
```diff
@@ -16,28 +15,19 @@
...
-     * Creates a new {@code SoapUserController}.
-     */
-    public SoapUserController() {
-        registry = new HashTokenRegistry();
-    }
+    private static final Logger logger = LogManager.getLogger(SoapUserEndpoint.class.getSimpleName());
 
     @Override
     public String login(@WebParam(name = "username") String username, @WebParam(name = "password") String password) {
         logger.debug("Soap request for login (username='{}', password='{}').", username, password);
```
Comment (ghost) [inline web/src/main/java/nl/tehdreamteam/se42/web/soap/user/SoapUserEndpoint.java:27]: BLOCKER Remove this hard-coded password. [rule](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AS2068)

**Comment by TehMessiah** [comment]
SonarQube analysis reported 7 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 4 major

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [SoapConversationEndpoint.java#L24](https://github.com/TehDreamTeam/SE42/blob/0072be0913de072d87faa7fb6fa8ae5099de0b7a/web/src/main/java/nl/tehdreamteam/se42/web/soap/conversation/SoapConversationEndpoint.java#L24): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [SoapConversationEndpoint.java#L30](https://github.com/TehDreamTeam/SE42/blob/0072be0913de072d87faa7fb6fa8ae5099de0b7a/web/src/main/java/nl/tehdreamteam/se42/web/soap/conversation/SoapConversationEndpoint.java#L30): Return an empty collection instead of null. [rule](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [SoapUserEndpoint.java#L37](https://github.com/TehDreamTeam/SE42/blob/0072be0913de072d87faa7fb6fa8ae5099de0b7a/web/src/main/java/nl/tehdreamteam/se42/web/soap/user/SoapUserEndpoint.java#L37): Assign this magic number 2L to a well-named constant, and use the constant instead. [rule](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [SoapUserEndpoint.java#L37](https://github.com/TehDreamTeam/SE42/blob/0072be0913de072d87faa7fb6fa8ae5099de0b7a/web/src/main/java/nl/tehdreamteam/se42/web/soap/user/SoapUserEndpoint.java#L37): Assign this magic number 4L to a well-named constant, and use the constant instead. [rule](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AS109)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by ghost** [comment]
SonarQube analysis reported 6 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [SoapConversationEndpoint.java#L24](https://github.com/TehDreamTeam/SE42/blob/f5dca2c66ef1e240f73c71bbd4d76713a327643e/web/src/main/java/nl/tehdreamteam/se42/web/soap/conversation/SoapConversationEndpoint.java#L24): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [SoapConversationEndpoint.java#L30](https://github.com/TehDreamTeam/SE42/blob/f5dca2c66ef1e240f73c71bbd4d76713a327643e/web/src/main/java/nl/tehdreamteam/se42/web/soap/conversation/SoapConversationEndpoint.java#L30): Return an empty collection instead of null. [rule](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [SoapUserEndpoint.java#L37](https://github.com/TehDreamTeam/SE42/blob/f5dca2c66ef1e240f73c71bbd4d76713a327643e/web/src/main/java/nl/tehdreamteam/se42/web/soap/user/SoapUserEndpoint.java#L37): Assign this magic number 2L to a well-named constant, and use the constant instead. [rule](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [SoapUserEndpoint.java#L37](https://github.com/TehDreamTeam/SE42/blob/f5dca2c66ef1e240f73c71bbd4d76713a327643e/web/src/main/java/nl/tehdreamteam/se42/web/soap/user/SoapUserEndpoint.java#L37): Assign this magic number 4L to a well-named constant, and use the constant instead. [rule](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AS109)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: web/src/main/java/nl/tehdreamteam/se42/web/soap/user/SoapUserEndpoint.java (hunk 3 above)
Comment (BeeHiveJava) [reply web/src/main/java/nl/tehdreamteam/se42/web/soap/user/SoapUserEndpoint.java:27]: Not a password.

File: web/src/main/java/nl/tehdreamteam/se42/web/soap/conversation/SoapConversationEndpoint.java (hunk 2 above)
Comment (BeeHiveJava) [reply web/src/main/java/nl/tehdreamteam/se42/web/soap/conversation/SoapConversationEndpoint.java:19]: Will be added in the future.

**Pull request merged**